level2/restaurants/gabi_restaurant.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class Restaurant
+  UNAVAILABLE_ITEM_ERROR = 'Items not available'
+  DOUBLE_DISCOUNT_ERROR = 'Double discount is not allowed'
+  attr_reader :name, :items
+
+  def initialize(name, *items)
+    @name = name.to_s
+    @items = items
+  end
+
+  def add_items(*items)
+    items.each { |i| @items << i }
+  end
+
+  def remove_item(item_name)
+    @items.delete_if { |item| item.name == item_name }
+  end
+
+  def place_order(ordered_items, discount: 0)
+    order_total = 0
+    items_in_stock = []
+
+    ordered_items.each do |ordered_item|
+      item = find_item(ordered_item)
+
+      return UNAVAILABLE_ITEM_ERROR if item.nil? || item_out_of_stock?(item, ordered_item)
```

Comment:
similar to the last suggestion, i think it would be better if the item itself was able to tell whether its available or not:
```suggestion
      return UNAVAILABLE_ITEM_ERROR unless item.available?(ordered_item[:count])
```

```
class Item
  def available?(count)
    ...
```